port_drayage_plugin/src/port_drayage_worker.cpp

```diff
@@ -91,61 +91,83 @@ namespace port_drayage_plugin
         return msg;
     }
 
-    void PortDrayageWorker::on_inbound_mobility_operation(const cav_msgs::MobilityOperationConstPtr& mobility_operation_msg)
-    {
+    void PortDrayageWorker::on_inbound_mobility_operation(const cav_msgs::MobilityOperationConstPtr& msg) {
         // Check if the received message is a new message for port drayage
-        if((mobility_operation_msg->strategy == PORT_DRAYAGE_STRATEGY_ID) && (mobility_operation_msg->strategy_params != _previous_strategy_params))
-        {
+        if((msg->strategy == PORT_DRAYAGE_STRATEGY_ID) && (msg->strategy_params != _previous_strategy_params)) {
             // Use Boost Property Tree to parse JSON-encoded strategy_params field in MobilityOperations message
             using boost::property_tree::ptree;
             ptree pt;
-            std::istringstream strategy_params_ss(mobility_operation_msg->strategy_params);
+            std::istringstream strategy_params_ss(msg->strategy_params);
             boost::property_tree::json_parser::read_json(strategy_params_ss, pt);
-            std::string received_cmv_id = pt.get<std::string>("cmv_id");
+            std::string mobility_operation_cmv_id = pt.get<std::string>("cmv_id");
 
             // Check if the received MobilityOperation message is intended for this vehicle's cmv_id   
-            if(received_cmv_id == _cmv_id)
-            {
-                ROS_DEBUG_STREAM("Processing new port drayage MobilityOperation message for cmv_id " << received_cmv_id);
-                _pdsm.process_event(PortDrayageEvent::RECEIVED_NEW_DESTINATION);
-                mobility_operation_message_parser(mobility_operation_msg->strategy_params);  
-                _previous_strategy_params = mobility_operation_msg->strategy_params;
+            if(mobility_operation_cmv_id == _cmv_id) {
+                ROS_DEBUG_STREAM("Processing new port drayage MobilityOperation message for cmv_id " << mobility_operation_cmv_id);
+                mobility_operation_message_parser(msg->strategy_params);  
+                _previous_strategy_params = msg->strategy_params;
+                
+                // Process event based on the PortDrayageEvent associated with the received MobilityOperation message
+                switch(_latest_mobility_operation_msg.port_drayage_event_type) {
+                    case PortDrayageEvent::RECEIVED_NEW_DESTINATION:
+                        _pdsm.process_event(PortDrayageEvent::RECEIVED_NEW_DESTINATION);
+                        break;
+                    default:
+                        break;
```

Comment:
Let's log that we hit this case and the actual received value here so that we can debug it easier if we run into issues, but otherwise this looks great to me!

Reply:
Added debug statements. Also fixed a small json-parsing bug that was causing an error in a unit test yesterday but I had originally missed it.